### DISCUSSION TITLE
"Paraphonic" switch?

### DISCUSSION THREAD

**Discussion by roguestreak** [marcel-licence/esp32_basic_synth#99] [Ideas]
Hi Marcel, I don't see this implemented but may have missed it; how hard would it be to add a "paraphonic" filter option?  ie., voices are polyphonic, but they all only get processed through the one filter?  This is _sometimes_ useful / cool if you want to do a sweeping filter across a lead line for example; as long as you overlap the key presses the filter envelope continues, only getting retriggered if keys don't overlap.  As an example, check out the bass synth from Nine Inch Nails' "Into The Void"
https://www.youtube.com/watch?v=G4lTMOmH8Dw
Could be the type of thing you'd only want active via a momentary button, hold down to enable during a lead break...
The "paraphonic" filter was both the biggest limitation and biggest uniqueness about the Korg Poly 800; start of this video shows it to a certain degree
https://www.youtube.com/watch?v=4qNWcSPYi7w

**Comment by marcel-licence** [comment]
Hi @roguestreak ,
I tried to look up the paraphonic function. I could imagine that a kind of legato function would also match your idea. There is a specific mono mode (which is not paraphonic) which prevents the envelope of retriggering when hitting a new key before releasing all keys.
Regarding the paraphonic, I've watched this video https://www.youtube.com/watch?v=n8yul4oNdLA and it sounds a bit confusing.

**Reply by roguestreak** [reply]
Hi Marcel, sorry it sounds like there are multiple definitions / implementations, that Behringer one is nothing like how the Poly 800 worked; I think "paraphonic" wasn't an official term for them, just a description by users, and it looks like I've muddied the waters by using that term.
Essentially the Poly 800 had polyphonic oscillators, but only a single filter they all went through, which is why people refer to it as being "para"-phonic rather than fully polyphonic.  The effect I'm asking about is potentially the "mono mode" you mention, however the mono only applies to the filter rather than the notes as well; does your current implementation allow this?  From previous videos I assumed mono dropped to singular notes, particularly when applying portamento.
If you look at the very start of the second video I posted it gives an example of playing overlapping polyphonic notes in such as way that the single filter isn't retriggered, all notes fall under the continuing envelope of the first trigger despite new notes being played.  So I guess it's a "mono filter mode", still allowing polyphony of voices but using mono mode on the filter only.

It was just a passing thought, as I'm looking at getting rid of my Poly 800 but this would be the one may thing I might miss about it.  On a side note, the DCO method is probably not very interesting but may be one to keep on a shelf of knowledge, it uses harmonics of squares to create saw waves a bit like your organ, some slight details in this overview vid (which also shows the mono filter, but in its "faults" manner)
https://www.youtube.com/watch?v=9puSeRrAXy4

Thanks again for all your work
Lee
